extensions/catalog/federated-catalog-cache/src/test/java/org/eclipse/dataspaceconnector/catalog/cache/TestProtocolAdapterRegistry.java

```diff
@@ -0,0 +1,50 @@
+package org.eclipse.dataspaceconnector.catalog.cache;
+
+import org.eclipse.dataspaceconnector.catalog.spi.CatalogQueryAdapter;
+import org.eclipse.dataspaceconnector.catalog.spi.CatalogQueryAdapterRegistry;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TestProtocolAdapterRegistry implements CatalogQueryAdapterRegistry {
```

Comment:
yep, they're very similar. the reason why I still added it is to avoid a "downward" dependency onto the in-memory one, even if it's a test dependency.